### ISSUE TITLE
Inconsistent implementations of IN4JSCore#findProject(nestedLocation)

### ISSUE THREAD

**Issue by mor-n4** [eclipse/n4js#934]
Method `IN4JSCore#findProject(URI)` has inconsistent implementations for the Eclipse and headless cases. When given an invalid location (i.e. a location for which no containing IN4JSProject exists) the method will return `Optional#absent()` in the headless case but will,  in the Eclipse case, return an instance of IN4JSProject for which `IN4JSProject#exists()` returns `false`.

Probably the behavior of the headless case is the correct one (when considering the API doc and the naming `findProject()`, esp. with regards to methods `findAllProjects()`, `findN4JSSourceContainers()`, etc.). But this should be double checked.

**Comment by szarnekow** [comment]
> Probably the behavior of the headless case is the correct one

True. Only IN4JSCore.create should return a project with `exists()==false`. The finders / methods with an optional return type, should yield absent instead.
